File: frontend/src/types/pet.ts
export type PetSpecies = 'DOG' | 'CAT' | 'RABBIT' | 'BIRD' | 'OTHER';
export type PetGender = 'MALE' | 'FEMALE' | 'UNKNOWN';

export const PET_SPECIES_OPTIONS: { value: PetSpecies; label: string }[] = [
  { value: 'DOG', label: '狗' },
  { value: 'CAT', label: '貓' },
  { value: 'RABBIT', label: '兔子' },
  { value: 'BIRD', label: '鳥' },
  { value: 'OTHER', label: '其他' },
];

export const PET_GENDER_OPTIONS: { value: PetGender; label: string }[] = [
  { value: 'MALE', label: '公' },
  { value: 'FEMALE', label: '母' },
  { value: 'UNKNOWN', label: '未知' },
];

export interface Pet {
  id: number;
  name: string;
  species: PetSpecies;
  breed: string | null;
  gender: PetGender | null;
  birthDate: string | null;
  color: string | null;
  notes: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface CreatePetRequest {
  name: string;
  species: PetSpecies;
  breed?: string | null;
  gender?: PetGender | null;
  birthDate?: string | null;
  color?: string | null;
  notes?: string | null;
}

export type UpdatePetRequest = Partial<CreatePetRequest>;

File: backend/src/main/java/com/harumi/petrecord/pet/Pet.java
package com.harumi.petrecord.pet;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.PrePersist;
import jakarta.persistence.PreUpdate;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.SQLDelete;
import org.hibernate.annotations.SQLRestriction;

import java.time.Instant;
import java.time.LocalDate;

@Entity
@Table(name = "pets")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@SQLDelete(sql = "UPDATE pets SET deleted_at = NOW() WHERE id = ?")
@SQLRestriction("deleted_at IS NULL")
public class Pet {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "owner_id", nullable = false)
    private Long ownerId;

    @Column(name = "name", nullable = false, length = 100)
    private String name;

    @Enumerated(EnumType.STRING)
    @Column(name = "species", nullable = false, length = 30)
    private PetSpecies species;

    @Column(name = "breed", length = 100)
    private String breed;

    @Enumerated(EnumType.STRING)
    @Column(name = "gender", length = 20)
    private PetGender gender;

    @Column(name = "birth_date")
    private LocalDate birthDate;

    @Column(name = "color", length = 50)
    private String color;

    @Column(name = "notes", columnDefinition = "TEXT")
    private String notes;

    @Column(name = "avatar_file_id")
    private Long avatarFileId;

    @Column(name = "created_at", nullable = false)
    private Instant createdAt;

    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    @Column(name = "deleted_at")
    private Instant deletedAt;

    @PrePersist
    void onCreate() {
        Instant now = Instant.now();
        if (createdAt == null) {
            createdAt = now;
        }
        updatedAt = now;
    }

    @PreUpdate
    void onUpdate() {
        updatedAt = Instant.now();
    }
}
